nixos/modules/services/monitoring/unifi-poller.nix

```diff
@@ -0,0 +1,227 @@
+          This may be noisy if you have a lot of devices. It adds one line per device.
+        '';
+      };
+      quiet = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Turns off per-interval logs. Only startup and error logs will be emitted.
+        '';
+      };
+      plugins = mkOption {
+        type = with types; listOf str;
+        default = [];
+        description = ''
+          Load additional plugins.
+        '';
+      };
+    };
+
+    prometheus = {
+      disable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to disable the prometheus ouput plugin.
+        '';
+      };
+      http_listen = mkOption {
+        type = types.str;
+        default = "0.0.0.0:9130";
+        description = ''
+          Bind the prometheus exporter to this IP or hostname.
+        '';
+      };
+      report_errors = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to report errors.
+        '';
+      };
+    };
+
+    influxdb = {
+      disable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to disable the influxdb ouput plugin.
+        '';
+      };
+      url = mkOption {
+        type = types.str;
+        default = "http://127.0.0.1:8086";
+        description = ''
+          URL of the influxdb host.
+        '';
+      };
+      user = mkOption {
+        type = types.str;
+        default = "unifipoller";
+        description = ''
+          Username for the influxdb.
+        '';
+      };
+      pass = mkOption {
+        type = types.str;
+        default = "unifipoller";
+        description = ''
+          Password for the influxdb.
+        '';
+      };
+      db = mkOption {
+        type = types.str;
+        default = "unifi";
+        description = ''
+          Database name. Database should exist.
+        '';
+      };
+      verify_ssl = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Verify the influxdb's certificate.
+        '';
+      };
+      interval = mkOption {
+        type = types.str;
+        default = "30s";
+        description = ''
+          Setting this lower than the Unifi controller's refresh
+          interval may lead to zeroes in your database.
+        '';
+      };
+    };
+
+    unifi = let
+      controllerOptions = {
+        user = mkOption {
+          type = types.str;
+          default = "unifi";
+          description = ''
+            Unifi service user name.
+          '';
+        };
+        pass = mkOption {
+          type = types.str;
+          default = "unifi";
+          description = ''
+            Unifi service user password.
+            If prefixed with "file://" the password will be read from the given file.
```

Comment:
Please add note about how the value should be have "file://" as a prefix to avoid storing secrets in the nix store. Additionally please add a check `warnings` which confirms that `cfg.unifi.pass` starts with `"file://"` and alerts the user they have made a terrible mistake otherwise.

Reply:
added the warnings. thanks for the hint!
what do you think about the updated description?

Reply:
unfortunately, implementing the warnings for the prometheus-exporter is less trivial due to how the module abstracts and simplifies adding new exporters. if you (or @WilliButz) have got any ideas on how to do this, i'd appreciate it :)

Reply:
Ah too bad, but I think you've taken sufficient steps if that's the case. Thanks!

Reply:
@elseym you can have warnings from inside the exporter submodules by using similar hack as the one for deprecated and replaced option names (see the `nginx` and `wireguard` exporter submodules), but so far this was never used.

Adding..
```nix
{
  /* ... */
  imports = [
    ({
      options.warnings = options.warnings;
      config.warnings = [ "some warning" ];
    })
  ];
}
```
..to the submodule should do the trick.

However, I would prefer having the password in a file by default, for example by using a Nix-specific option called `passwordFile` which takes a canonical path that needs to be accessible by the unifi-poller and is internally converted to `pass` and prefixed with `file://` 
This should only cause secrets to be placed in the Nix store, if it is done on purpose by using something like `writeText`.
(Of course this would deviate from the configuration used on other systems, but these don't usually make config files world-readable ^^)

Also, a note about the password file needing to be accessible by the service would probably help with potential issues regarding the deployment of remote hosts in any of the two scenarios.

Reply:
aaah imports 😄 didn't think of that. thanks!
i got rid of the need for warnings now by only allowing password files. see https://github.com/NixOS/nixpkgs/pull/96830#issuecomment-687828987.